core/jvm/test/InstantParsing.kt

```diff
@@ -0,0 +1,37 @@
+package kotlinx.datetime
+
+import kotlinx.datetime.format.*
+import kotlin.test.*
+
+class InstantParsing {
+    @Test
+    fun testParsingInvalidInstants() {
+        fun parseInstantLikeJavaDoes(input: String): Instant =
+            DateTimeComponents.Formats.ISO_DATE_TIME_OFFSET.parse(input).apply {
+                when {
+                    hour == 24 && minute == 0 && second == 0 && nanosecond == 0 -> {
+                        setDate(toLocalDate().plus(1, DateTimeUnit.DAY))
+                        hour = 0
+                    }
+                    hour == 23 && minute == 59 && second == 60 -> second = 59
+                }
+            }.toInstantUsingOffset()
+        fun formatTwoDigits(i: Int) = if (i < 10) "0$i" else "$i"
+        for (hour in 23..25) {
+            for (minuteRange in listOf(0..5, 58..62)) {
+                for (minute in minuteRange) {
+                    for (secondRange in listOf(0..5, 58..62)) {
+                        for (second in secondRange) {
+                            val input = "2020-03-16T${hour}:${formatTwoDigits(minute)}:${formatTwoDigits(second)}Z"
+                            assertEquals(
+                                runCatching { java.time.Instant.parse(input) }.getOrNull(),
+                                kotlin.runCatching { parseInstantLikeJavaDoes(input).toJavaInstant() }.getOrNull()
```

Comment:
Strange that here we need to use runCatching qualified.

Reply:
We don't, IDEA did this for some reason during a refactoring.